main.tf

```diff
@@ -19,24 +19,25 @@ resource "aws_ec2_transit_gateway" "default" {
 }
 
 resource "aws_ec2_transit_gateway_route_table" "default" {
-  count              = var.create_transit_gateway_route_table ? 1 : 0
+  count              = module.this.enabled && var.create_transit_gateway_route_table ? 1 : 0
   transit_gateway_id = local.transit_gateway_id
   tags               = module.this.tags
 }
 
 # Need to find out if VPC is in same account as Transit Gateway.
 # See resource "aws_ec2_transit_gateway_vpc_attachment" below.
 data "aws_ec2_transit_gateway" "this" {
-  id = local.transit_gateway_id
+  count = (module.this.enabled && ((var.existing_transit_gateway_id != null && var.existing_transit_gateway_id != "") || var.create_transit_gateway)) ? 1 : 0
```

Comment:
please review this logic for the case when `module.this.enabled=false` and `var.existing_transit_gateway_id=true`

Reply:
if `module.this.enabled=false` there's a test which is passing. The whole expression evaluates to false, regardless of the other variables.

If `module.this.enabled=true ` and `var.create_transit_gateway=true`, there's a test for this which is passing. The whole expression evaluates to true, regardless of the other variable (which is false since it's null).

If `module.this.enabled=true ` and `var.existing_transit_gateway_id=tgw-12345`, the second bracket expression will evaluate to true since (var.existing_transit_gateway_id != null && var.existing_transit_gateway_id != "") is true. Hence the whole expression evaluates to true. There is no test for this, but I use this feature in my root module and it works as expected.

The TF order of operations has && at a higher precedence than || but I added the brackets to be clear about it to the reader.

Note: this is the same logic as `local.transit_gateway_id`. I tried to reuse that originally but you get the dreaded error: `"count" value depends on resource attributes that cannot be determined until apply`
See 5a597958.